beest/cmd/driver/terraform.go

```diff
@@ -0,0 +1,192 @@
+package driver
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-config-inspect/tfconfig"
+	"github.com/hashicorp/terraform-exec/tfexec"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+)
+
+const (
+	DefaultTerraformBinary = "/usr/local/bin/terraform"
+)
+
+type TerraformContext interface {
+	WorkingDir() string
+	Variables() map[string]interface{}
+}
+
+type TerraformState struct {
+	Module   *tfconfig.Module
+	tf       *tfexec.Terraform
+	planPath string
+}
+
+func TerraformApply(ctx TerraformContext, prompt bool) {
+	apply(ctx, false, prompt)
+}
+
+func TerraformDestroy(ctx TerraformContext, prompt bool) {
+	apply(ctx, true, prompt)
+}
+
+func apply(ctx TerraformContext, destroy bool, prompt bool) {
+	state := newTerraform(ctx)
+	state.init()
+	state.validate()
+	log.Println(fmt.Sprintf("Variables: %v", ctx.Variables()))
+	if state.plan(ctx.Variables(), destroy) {
+		state.apply(prompt)
+	} else {
+		log.Println("No Terraform changes detected.")
+	}
+}
+
+func newTerraform(ctx TerraformContext) *TerraformState {
+	log.Println(fmt.Sprintf("Preparing Terraform to run from: %s", ctx.WorkingDir()))
+
+	tf, err := tfexec.NewTerraform(ctx.WorkingDir(), DefaultTerraformBinary)
+	if err != nil {
+		log.Fatalf("Error running NewTerraform: %s", err)
+	}
+
+	// TODO debug
```

Comment:
expand this TODO so we know what is your idea

Reply:
I'll remove and created a ticket to introduce a CLI flag to output debug information https://stackstate.atlassian.net/browse/STAC-15551